### ISSUE TITLE
DAM model in serving

### ISSUE THREAD

**Issue by szho42** [baidu/Dialogue#11]
Hi, authors,

In the production stage, given a fix size of canned responses,  it needs to compute the confidence level, i.e. logits, with each canned responses. Do you guys have a smarter way of getting the top n from the canned responses, for each context?

Cheers

**Comment by xyzhou-puck** [comment]
Hi,

I guess you may want to fetch top-n responses from m candidates, where m >> n. 

If yes, you can try a min-heap with size n and iterate all candidates, this would be much faster than sorting all elements.

Xiangyang

**Comment by xyzhou-puck** [comment]
Thanks so much and you are welcome to reach out us for any questions.

Xiangyang
